api/src/main/java/com/viaversion/viaversion/api/data/MappingDataLoader.java

```diff
@@ -51,17 +51,20 @@
 import java.util.stream.Collectors;
 import org.checkerframework.checker.nullness.qual.Nullable;
 
-public final class MappingDataLoader {
+public class MappingDataLoader {
 
-    private static final Map<String, CompoundTag> MAPPINGS_CACHE = new HashMap<>();
-    private static final TagReader<CompoundTag> MAPPINGS_READER = NBTIO.reader(CompoundTag.class).named();
     private static final byte DIRECT_ID = 0;
     private static final byte SHIFTS_ID = 1;
     private static final byte CHANGES_ID = 2;
     private static final byte IDENTITY_ID = 3;
-    private static boolean cacheValid = true;
 
-    public static void clearCache() {
+    public static final MappingDataLoader INSTANCE = new MappingDataLoader();
+
+    private final Map<String, CompoundTag> MAPPINGS_CACHE = new HashMap<>();
+    private final TagReader<CompoundTag> MAPPINGS_READER = NBTIO.reader(CompoundTag.class).named();
```

Comment:
Mappings cache needs to be renamed, and mappings reader can stay static

Reply:
Fixed